tests/playbooks/tests_bond.yml

```diff
@@ -0,0 +1,81 @@
+# SPDX-License-Identifier: BSD-3-Clause
+---
+- hosts: all
+  vars:
+    master_profile: bond0
+    master_device: nm-bond
+    slave1_profile: bond0.0
+    slave1_device: test1
+    slave2_profile: bond0.1
+    slave2_device: test2
+  tasks:
+    - name: "INIT Prepare setup"
+      debug:
+        msg: "##################################################"
+    - include_tasks: tasks/create_test_interfaces.yml
+    - include_tasks: tasks/assert_device_present.yml
+      vars:
+        interface: "{{ slave1_device }}"
+    - include_tasks: tasks/assert_device_present.yml
+      vars:
+        interface: "{{ slave2_device }}"
+    - block:
+        - name: "TEST Add Bond with 2 slaves"
+          debug:
+            msg: "##################################################"
+        - import_role:
+            name: linux-system-roles.network
+          vars:
+            network_connections:
+              # Create a bond master
+              - name: "{{ master_profile }}"
+                state: up
+                type: bond
+                interface_name: "{{ master_device }}"
+                # bond configuration settings: (optional)
+                bond:
+                  mode: active-backup
+                  miimon: 110
+              # enslave an ethernet to the bond
+              - name: "{{ slave1_profile }}"
+                state: up
+                type: ethernet
+                interface_name: "{{ slave1_device }}"
+                master: "{{ master_profile }}"
+              # enslave a second ethernet to the bond
+              - name: "{{ slave2_profile }}"
+                state: up
+                type: ethernet
+                interface_name: "{{ slave2_device }}"
+                master: "{{ master_profile }}"
+        - include_tasks: tasks/assert_device_present.yml
+          vars:
+            interface: "{{ master_device }}"
+        - include_tasks: tasks/assert_profile_present.yml
+          vars:
+            profile: "{{ item }}"
+          loop:
+            - "{{ master_profile }}"
+            - "{{ slave1_profile }}"
+            - "{{ slave2_profile }}"
+
+      always:
+        - block:
+            - import_role:
```

Comment:
The cleanup could be simplified by running the module instead importing the role:
https://github.com/linux-system-roles/network/blob/master/tests/playbooks/tests_checkpoint_cleanup.yml#L70

Reply:
am I able to remove namespace?

Reply:
ok, I am, but when I am creating such a setup I think there should be cleanup too so one doesn't need to copy paste 20 lines of code. 
https://docs.ansible.com/ansible/latest/modules/ip_netns_module.html